### PR TITLE
[PLATFORM-2332]: Implement caching using dynamo

### PR DESCRIPTION
**Pull request by MaeIsBad**

### CODE REVIEW

File: test/prima_auth0_ex/token_cache/dynamodb_test.exs [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+    # Token expiration is managed by aws, and could take days for old tokens to be deleted,
+    # so we don't cover that in the tests here
+    test_token_expiration: false
+
+  setup do
+    cache_env = Application.get_env(:prima_auth0_ex, :dynamodb_cache)
+
+    on_exit(fn ->
+      if cache_env == nil do
+        Application.delete_env(:prima_auth0_ex, :dynamodb_cache)
```
Comment (cpiemontese) [inline test/prima_auth0_ex/token_cache/dynamodb_test.exs:16]: Why delete if nil...? I don't understand the logic

File: lib/prima_auth0_ex/token_cache/dynamodb.ex [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+  def create_table? do
+    Config.dynamodb(:create_table, true)
+  end
+
+  defp table_name do
+    Config.dynamodb!(:table_name)
+  end
+
+  def key(client \\ :default_client, audience) do
+    "#{client}:#{audience}"
```
Comment (cpiemontese) [inline lib/prima_auth0_ex/token_cache/dynamodb.ex:103]: In redis we use a [specific format](https://github.com/primait/auth0_ex/blob/master/lib/prima_auth0_ex/token_cache/encrypted_redis_token_cache.ex#L60), mainly due to the fact that the cache was shared with other services. I assume we don't need to do anything specific for Dynamo and what you wrote suffices

**Review by cpiemontese — APPROVED** [review]
Really really nice! You manage to refactor tests too... :eyes:

File: config/dev.exs [hunk 3]
```diff
@@ -3,3 +3,13 @@ import Config
...
 config :logger, level: :debug
+
+config :ex_aws,
+  access_key_id: "ABCD",
+  secret_access_key: "secret"
+
+config :ex_aws, :dynamodb,
+  scheme: "http://",
+  host: "dynamodb",
+  port: 8000,
```
[the remark this answers is not in the document]
Comment (MaeIsBad) [reply config/dev.exs:14]: oh yeah

File: test/prima_auth0_ex/token_cache/dynamodb_test.exs (hunk 1 above)
Comment (MaeIsBad) [reply test/prima_auth0_ex/token_cache/dynamodb_test.exs:16]: if it was nil, before we saved it we just reset it

File: lib/prima_auth0_ex/token_cache/dynamodb.ex (hunk 2 above)
Comment (MaeIsBad) [reply lib/prima_auth0_ex/token_cache/dynamodb.ex:103]: yeah as long as we keep the same one for all other services as well

Comment (MaeIsBad) [reply lib/prima_auth0_ex/token_cache/dynamodb.ex:103]: I'll add a prefix, just in case

Comment (MaeIsBad) [reply lib/prima_auth0_ex/token_cache/dynamodb.ex:103]: We don't need to do that here, because `table_name` is configurable

**Review by cpiemontese — APPROVED** [review]

**Review by cpiemontese — APPROVED** [review]

**Pull request merged**